### PR TITLE
Paint control tweaks

### PR DESCRIPTION
**Pull request by dylwhich**
* Adds option to hold the touchpad (anywhere) instead of select to change colors/brushes
* In palette select mode, swaps functions of D-pad left/right and select. Now, d-pad picks color channel and select increases the color channel value
* Also in palette select mode, the touchpad now sets the color channe value based on the absolute touch position centroid
* Pressing A will always draw at least once, even if you press & release before the next frame
* Moved flood fill to bresenham.h

### CODE REVIEW

File: main/modes/paint/paint_draw.c [hunk 1]
```diff
@@ -1120,44 +1139,73 @@ void paintDrawScreenTouchCb(const touch_event_t* evt)
...
             }
+
+            break;
         }
-        paintState->touchDown = false;
+
+        case BTN_MODE_PALETTE:
+        {
+            int32_t centroid, intensity;
+            if (getTouchCentroid(&centroid, &intensity))
```
Comment (AEFeinstein) [inline main/modes/paint/paint_draw.c:1197]: This doesn't work the way you think it works. The callback is only called when a pad is touched or released, but for maximum smoothness, this should be called somewhere else and more frequently (`paintMainLoop()` perhaps). I want to see the brush modifies (x1, x2, x3) update in realtime as I swipe!

**Comment by AEFeinstein** [comment]
Make sure to update the tutorial with these changes too

File: main/modes/paint/paint_draw.c (hunk 1 above)
Comment (dylwhich) [reply main/modes/paint/paint_draw.c]: heh, I thought maybe I could get away with it like this but yeah it does make it very choppy...

Comment (dylwhich) [reply main/modes/paint/paint_draw.c]: Now it's suuuuper smooth! And the brush size updates in real-time too

Comment (AEFeinstein) [reply main/modes/paint/paint_draw.c]: Much better!

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**